Harmonize options for parameter technology across api and bulk

chrwm opened the pull request:
#252 

Harmonizes options for parameter technology across api and bulk download.

chrwm committed:
Add quick solution for 'location' #252

FlorianK13 reviewed (approved):
Seems good

chrwm commented:
Hi Florian, thanks for the initial review. I think the pull request is not quite ready yet for review. I just implemented the input `location`. As listed in #252, I still need and will implement in this PR the inputs:

- electricity_consumer
- balancing_area
- market
- permit
- gas
- grid

I just meant to signal that I'm working on the topic by opening the draft-pull request.

chrwm committed:
Relocate technology_input_harmonisation function to helpers #252

* and apply black

chrwm committed:
Harmonise technology parameter "permit" #252

chrwm committed:
Split harmonisation and printing of api settings in two functions #252

* Split harmonisation and printing of api settings in two functions for better maintenance
* Add harmonisation of technology parameter "permit"

chrwm committed:
Fix location harmonisation #252

Return all api_location_types in variable "locs" and sets them as new api_location_types if "location" is set in technology

chrwm committed:
Harmonise technology parameter "electricity_consumer" #252

* and delete "consumer" from bulk_technologies in _validate_parameter_format_for_download_method, it was artifact from testing

chrwm committed:
Simplify location harmonisation #252

* delete unnecessary if-statement
* api_location_types will be overwritten when location is in technology

chrwm committed:
Merge branch 'develop' into feature-252-technology-parameter-api-bulk

chrwm commented:
I can't keep up with the current development speed, so we better merge this now or not.

FlorianK13 commented on open_mastr/utils/helpers.py:
```diff
@@ -249,6 +250,10 @@ def raise_warning_for_invalid_parameter_combinations(
 
+
+def db_engine():  # TODO: Include in _create_database in MaStR() class
+    return create_engine(DB_URL)
```
@deniztepe If I remember correct you deleted that function, or am I wrong?

deniztepe replied:
Yes, it seems like that these 3 lines are slipped in during a merge.

FlorianK13 replied:
I deleted it again

FlorianK13 commented on open_mastr/utils/helpers.py:
```diff
@@ -282,3 +287,73 @@ def session_scope(engine):
         session.close()
+
+
+def technology_input_harmonisation(technology, api_data_types):
```
Maybe we could put that function to helpers.py ?

deniztepe replied:
It is in helpers.py, isn't it?

FlorianK13 replied:
Ah sorry, I got confused.

FlorianK13 committed:
Delete db_engine function #252

The function reappered during some merging, but it
should be deleted.

Additionally sort imported packages and apply black.

chrwm commented:
Thanks for the review! Apart from the current reviews and changes, can I go ahead and merge? @deniztepe it seems you haven't approved yet.

deniztepe commented:
Seems good for me too. Sorry for delay.

chrwm commented:
No worries, thank you very much!
